save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt

```diff
@@ -227,33 +229,63 @@ class WarnPlugin(
      * 1) reading the file
      * 2) for each line get the warning
      */
+    @Suppress("TOO_LONG_FUNCTION")
     private fun Path.collectWarningsWithLineNumbers(
         warnPluginConfig: WarnPluginConfig,
         generalConfig: GeneralConfig
     ): List<Warning> {
         val linesFile = fs.readLines(this)
-        return linesFile
-            .mapIndexed { index, line ->
-                val newLine = line.getLineNumber(
+        return generalConfig.expectedWarningsEndPattern?.let {
```

Comment:
May be you can extract this logic into two functions, instead of two large blocks?

Reply:
Yes, I think it would be better :) Fixed

Reply:
Yes, `collectWarningsWithLineNumbers` seems to complicated